graphql-api/src/graphql/graphql-api.ts

```diff
@@ -52,7 +52,24 @@ const customValidateFn = (...args: Parameters<typeof validate>) => {
   )
 }
 
-const customFormatErrorFn = (error: any, request: any, graphqlRequestParams: any) => {
+const formatErrorAndSetNocache = (
+  error: any,
+  request: any,
+  graphqlRequestParams: any,
+  response: any
+) => {
+  // Setting the nocache header in this function that's just supposed to format
+  // an error is pretty gross, but given the multitude of places that this
+  // API can return or throw an error, this is the only obvious spot where we
+  // can intercept all errors. We can't just wrap the entire API in a try/catch
+  // either and set the header in a catch, since after the API's run we've
+  // already sent the response.
+  //
+  // Also, as written, this will no-cache not-found errors, which is
+  // undesirable, but something we can put off fixing until later.
+
+  response.set('Cache-Control', 'no-cache')
```

Comment:
Per [this Slack thread](https://the-tgg.slack.com/archives/C03P7FA3W3T/p1698264290656549?thread_ts=1698261608.539069&cid=C03P7FA3W3T), I believe this should be `no-store` rather than `no-cache`, to have our desired effect

Reply:
Yes thank you nice catch